File: app/src/main/java/io/goooler/demoapp/util/SpUtil.java
package io.goooler.demoapp.util;

import android.content.SharedPreferences;

import io.goooler.demoapp.base.BaseApplication;
import io.goooler.demoapp.model.Constants;

import static android.content.Context.MODE_PRIVATE;

/**
 * SharedPreferences 简单封装
 */

public class SpUtil {

    /**
     * 获取 sp
     *
     * @param spName sp 文件名
     */
    public static SharedPreferences getSp(String spName) {
        return BaseApplication.getContext().getSharedPreferences(spName, MODE_PRIVATE);
    }

    /**
     * 获取 sp edit
     *
     * @param spName sp 文件名
     */
    public static SharedPreferences.Editor getSpEdit(String spName) {
        return getSp(spName).edit();
    }

    /**
     * 设置应用是否第一次启动的状态
     */
    public static void setFirstRunState(boolean firstRunState) {
        getSpEdit(Constants.SP_RUNINFO).putBoolean(Constants.SP_FIRST_RUN, firstRunState).apply();
    }

    /**
     * 判断应用是否第一次启动
     */
    public static boolean isFirstRun() {
        return getSp(Constants.SP_RUNINFO).getBoolean(Constants.SP_FIRST_RUN, true);
    }
}

File: app/src/main/java/io/goooler/demoapp/util/SpUtil.kt
package io.goooler.demoapp.util

import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import io.goooler.demoapp.base.BaseApplication
import io.goooler.demoapp.model.SpKeys

/**
 * SharedPreferences 简单封装
 */

object SpUtil {

    private const val SP_CONFIG = "sp_config"

    /**
     * 判断应用是否第一次启动
     */
    val isFirstRun = getSp().getBoolean(SpKeys.SP_FIRST_RUN, true)

    /**
     * 设置应用是否第一次启动的状态
     */
    fun setFirstRunState(state: Boolean = false) {
        putBoolean(SpKeys.SP_FIRST_RUN, state)
    }

    fun getBoolean(@SpKeys spKey: String, default: Boolean = false): Boolean {
        return getSp().getBoolean(spKey, default)
    }

    fun putBoolean(@SpKeys spKey: String, value: Boolean = true) {
        getSp().edit().putBoolean(spKey, value).apply()
    }

    /**
     * 获取 sp
     *
     * @param spName sp 文件名
     */
    private fun getSp(spName: String = SP_CONFIG): SharedPreferences {
        return BaseApplication.context.getSharedPreferences(spName, MODE_PRIVATE)
    }
}
